charts/matrix-stack/templates/deployment-markers/role.yaml

```diff
@@ -0,0 +1,35 @@
+{{- /*
+Copyright 2025 New Vector Ltd
+
+SPDX-License-Identifier: AGPL-3.0-only
+*/ -}}
+{{ range $step := list "pre" "post" -}}
+{{- with $.Values.deploymentMarkers -}}
+{{- if and .enabled .rbac.create (include "element-io.deployment-markers.secret-markers" (dict "root" $ "context" (dict "step" $step))) }}
+apiVersion: rbac.authorization.k8s.io/v1
+kind: Role
+metadata:
+  name: {{ $.Release.Name }}-deployment-markers-{{ $step }}
+  namespace: {{ $.Release.Namespace }}
+  labels:
+    {{- include "element-io.deployment-markers.labels" (dict "root" $ "context" (mustMergeOverwrite (dict "step" $step) .)) | nindent 4 }}
+  annotations:
+    "helm.sh/hook": {{ $step }}-install,{{ $step }}-upgrade
+    "helm.sh/hook-weight": "-20"
+rules:
+{{/*
+  https://kubernetes.io/docs/reference/access-authn-authz/rbac/#referring-to-resources
+  You cannot restrict create or deletecollection requests by resourceName.
+  For create, this limitation is because the object name is not known at authorization time.
+*/}}
+- apiGroups: [""]
+  resources: ["secrets"]
```

Comment:
Is the reason to use `Secrets` so that we can reuse the existing `matrix-tools` infra for writing to `Secrets`? Because there's nothing secret/credentially about these markers, so would naively have put them in `ConfigMap`

Reply:
Yes it's mainly the reason. But I thought about it last night and I think I will use another infra, as to make sure we do not confuse both tools (potentially we could generate a marker in the `init-secrets` container and it just feels wrong)